### ISSUE TITLE
Over-eager zone matching

### ISSUE THREAD

**Issue by bigdogwillfeed** [nodecraft/acme-dns-01-cloudflare#16]
Thanks for the plugin! I've been able to get it up and running with Greenlock with minimal trouble. However, I did hit a roadblock that I would really appreciate some help with.

We own the domains `devresults.com` and `training-devresults.com`, but I'm not able to generate certificates for `training-devresults.com` because of the logic in how the zone is picked from the list of zones thinks that `devresults.com` is the zone for e.g., `*.training-devresults.com`:
https://github.com/nodecraft/acme-dns-01-cloudflare/blob/450850f5d649ce465dafb7034380685a6f46090b/index.js#L228

I made the zone check a bit more strict, and the following works for my in my dev environment (check for exact domain or subdomain match):
```js
if(domain === zone.name || domain.endsWith(`.${zone.name}`)){
```

Can you help me out?

**Comment by Cherry** [comment]
Thanks for the issue!

This definitely isn't something we thought about during testing, so I appreciate you bringing it to light. Your suggested change looks good - I'll see about testing that and implementing a fix for this soon. 👍

**Comment by Cherry** [comment]
The fix for landed in version 1.2.3. Thanks for the report once again!

**Comment by bigdogwillfeed** [comment]
Works great; thanks for the speedy fix!!
